packages/app-desktop/gui/NoteEditor/utils/contextMenuUtils.ts

```diff
@@ -0,0 +1,53 @@
+import Resource from '@joplin/lib/models/Resource';
+
+export enum ContextMenuItemType {
+	None = '',
+	Image = 'image',
+	Resource = 'resource',
+	Text = 'text',
+	Link = 'link',
+}
+
+export interface ContextMenuOptions {
+	itemType: ContextMenuItemType;
+	resourceId: string;
+	tempResource: {
+		data: any; // string or Uint8Array
+		mime: string;
+		filename: string;
+	};
+	linkToCopy: string;
+	textToCopy: string;
+	htmlToCopy: string;
+	insertContent: Function;
+	isReadOnly?: boolean;
+	isTemp?: boolean;
+}
+
+export interface ContextMenuItem {
+	label: string;
+	onAction: Function;
+	isActive: Function;
+}
+
+export interface ContextMenuItems {
+	[key: string]: ContextMenuItem;
+}
+
+export async function resourceInfo(options: ContextMenuOptions): Promise<any> {
+	const resource = options.resourceId ? await Resource.load(options.resourceId) : null;
+	const filePath = resource ? Resource.fullPath(resource) : null;
+	const filename = resource ? (resource.filename ? resource.filename : resource.title) : options.tempResource && options.tempResource.filename ? options.tempResource.filename : '';
+	const getCopyPath = () => {
+		if (options.tempResource) {
+			return tempResourceToDataUri(options.tempResource);
+		}
+		return filePath;
+	};
```

Comment:
Please remove that from here.

Reply:
Actually don't change this one at all. It's used to get the resource info and nothing else.